gst/nnstreamer/tensor_filter/tensor_filter_custom_easy.c

```diff
@@ -66,15 +66,19 @@ NNS_custom_easy_register (const char *modelname,
   if (!func || !in_info || !out_info)
     return -EINVAL;
 
+  if (!gst_tensors_info_validate (in_info) ||
+      !gst_tensors_info_validate (out_info))
+    return -EINVAL;
+
   ptr = g_new0 (internal_data, 1);
 
   if (!ptr)
     return -ENOMEM;
 
   ptr->func = func;
   ptr->data = data;
-  ptr->in_info = in_info;
-  ptr->out_info = out_info;
+  gst_tensors_info_copy (&ptr->in_info, in_info);
+  gst_tensors_info_copy (&ptr->out_info, out_info);
```

Comment:
I think you need to call `gst_tensors_info_free` as well in error handling (e.g., before `g_free (ptr)` in line 86).

Reply:
You are right. I added a function to free internal data.
When unregistering custom filter, it is also necessary to free internal data.